addons/payment_nuvei/tests/test_payment_transaction.py

```diff
@@ -0,0 +1,146 @@
+            'payment_method_mode': 'filter',
+            'payment_method': 'unknown',
+            'phone1': phone_number,
+            'state': tx.partner_state_id.code or '',
+            'time_stamp': tx.create_date.strftime('%Y-%m-%d.%H:%M:%S'),
+            'total_amount': self.amount,
+            'version': '4.0.0',
+            'zip': tx.partner_zip,
+            'notify_url': webhook_url,
+            'success_url': return_url,
+            'error_url': f'{return_url}?{urls.url_encode(cancel_url_params)}',
+            'pending_url': return_url,
+            'back_url': f'{return_url}?{urls.url_encode(cancel_url_params)}',
+        }
+
+    def test_no_item_missing_from_rendering_values(self):
+        """ Test that the rendered values are conform to the transaction fields. """
+
+        tx = self._create_transaction(flow='redirect')
+        with patch(
+            'odoo.addons.payment.utils.generate_access_token', new=self._generate_test_access_token
+        ):
+            processing_values = tx._get_specific_rendering_values(None)
+        expected_values = self._get_expected_values(tx)
+        self.assertDictEqual(processing_values['url_params'], expected_values)
+
+    @mute_logger('odoo.addons.payment.models.payment_transaction')
+    def test_no_input_missing_from_redirect_form(self):
+        """ Test that the no key is not omitted from the rendering values. """
+        tx = self._create_transaction(flow='redirect')
+        expected_input_keys = [
+            'checksum',
+            'address1',
+            'city',
+            'country',
+            'currency',
+            'email',
+            'encoding',
+            'first_name',
+            'item_amount_1',
+            'item_name_1',
+            'item_quantity_1',
+            'invoice_id',
+            'last_name',
+            'merchantLocale',
+            'merchant_id',
+            'merchant_site_id',
+            'payment_method_mode',
+            'payment_method',
+            'phone1',
+            'state',
+            'time_stamp',
+            'total_amount',
+            'version',
+            'zip',
+            'notify_url',
+            'success_url',
+            'error_url',
+            'pending_url',
+            'back_url',
+        ]
+        expected_input_keys.sort()
+        with patch(
+            'odoo.addons.payment.utils.generate_access_token', new=self._generate_test_access_token
+        ):
+            processing_values = tx._get_processing_values()
+
+        form_info = self._extract_values_from_html_form(processing_values['redirect_form_html'])
+        self.assertEqual(form_info['action'], 'https://ppp-test.safecharge.com/ppp/purchase.do')
+        self.assertEqual(form_info['method'], 'post')
+        input_keys = list(form_info['inputs'].keys())
+        input_keys.sort()
+        self.assertListEqual(input_keys, expected_input_keys)
+
+    def test_processing_notification_data_confirms_transaction(self):
```

Comment:
You can also test with the other states that are implemented in your `_process_notification_data`, for instance:
- `test_processing_notification_data_sets_transaction_in_error`